lightning-invoice/src/utils.rs

```diff
@@ -326,14 +341,15 @@ where
 	F::Target: FeeEstimator,
 	L::Target: Logger,
 {
-	let route_hints = filter_channels(channelmanager.list_usable_channels(), amt_msat);
-
 	// `create_inbound_payment` only returns an error if the amount is greater than the total bitcoin
 	// supply.
 	let (payment_hash, payment_secret) = channelmanager
 		.create_inbound_payment(amt_msat, invoice_expiry_delta_secs)
 		.map_err(|()| SignOrCreationError::CreationError(CreationError::InvalidAmount))?;
 	let our_node_pubkey = channelmanager.get_our_node_id();
+	let channels = channelmanager.list_usable_channels();
```

Comment:
Are we using this value other than in +L373? Should we call it inline in that call like -L329?

Reply:
I moved it to keep all the `ChannelManager` calls together. Note that `our_node_pubkey` is only used once, too.